lib/opal/rewriters/base.rb

```diff
@@ -94,6 +95,22 @@ def append_to_body(body, node)
           s(:begin, body, node)
         end
       end
+
+      # Store the current node for reporting.
+      attr_accessor :current_node
+
+      # Intercept the main call and assign current node.
+      def process(node)
+        self.current_node = node
+        super
+      end
```

Comment:
Probably it makes sense to add `ensure; self.current_node = nil` to prevent confusing error locations when someone calls `on_*` manually.